CMakeLists.txt

```diff
@@ -18,10 +18,87 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 3.3)
         message(FATAL_ERROR "Clang version must be at least 3.3!")
     endif()
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC")
+    # allow MSVC VS2015.1 and VS2013 toolchain ( VS2013 is not tested )
+    if ("${CMAKE_CXX_COMPILER_VERSION}" VERSION_EQUAL "19.0")
+        if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_EQUAL "19.0.23506.0")
+            message(FATAL_ERROR "Your MSVC version ${CMAKE_CXX_COMPILER_VERSION} does not match the required 19.0.23506.0 version ( Visual Studio 2015 Update 1 ).")
```

Comment:
The message is wrong I think - according to the code we do support other versions, but not other 19.0.x versions.

Reply:
Yes, you got it right :)

I did not want to be very restrictive, but I know from trial and error that for Visual Studio 2015 ( compiler version 19.0 ) the only working version is the specified one.
I have not tried older compilers, but I have a suspicion that it will not work due to missing C++11 features.